### PR TITLE
[system-probe] Enable dnat tests for protocol classification

### PR DESCRIPTION
**Pull request by hmahmood**
### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/tracer_classification_test.go [hunk 1]
```diff
@@ -478,10 +479,6 @@ func testPostgresProtocolClassification(t *testing.T, cfg *config.Config, client
 		targetAddress: targetHost,
 	})
 
-	if clientHost != "127.0.0.1" {
-		t.Skip("postgres tests are not supported DNat")
-	}
```
Comment (guyarb) [inline pkg/network/tracer/tracer_classification_test.go:483]: revert, this test cannot run on NAT (regardless of your fix)

File: pkg/network/tracer/tracer_classification_test.go [hunk 2]
```diff
@@ -830,7 +827,7 @@ func testMongoProtocolClassification(t *testing.T, cfg *config.Config, clientHos
 }
 
 func testRedisProtocolClassification(t *testing.T, cfg *config.Config, clientHost, targetHost, serverHost string) {
-	skipFunc := composeSkips(skipIfNotLinux, skipIfUsingNAT)
+	skipFunc := skipIfNotLinux
```
Comment (guyarb) [inline pkg/network/tracer/tracer_classification_test.go:830]: revert, this test cannot run on NAT (regardless of your fix)

**Review by guyarb — CHANGES_REQUESTED** [review]

File: pkg/network/tracer/tracer_classification_test.go [hunk 3]
```diff
@@ -1419,23 +1416,7 @@ func waitForConnectionsWithProtocol(t *testing.T, tr *Tracer, targetAddr, server
...
-
-		if !foundOutgoingWithProtocol && len(outgoingConns) == 0 {
-			t.Log("npm didn't find the outgoing connections, not failing the test")
-			return
-		}
-
-	}
-
-	t.Errorf("couldn't find incoming and outgoing connections with protocol %d for "+
+	assert.Failf(t, "", "couldn't find incoming and outgoing connections with protocol %d for "+
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/tracer/tracer_classification_test.go:1423]: yes

**Review by guyarb — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

**Comment by guyarb** [comment]
The PR didn't trigger kitchen tests, triggering them manually

**Review by guyarb — CHANGES_REQUESTED** [review]
CI is failing
https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/232833493

**Comment by hmahmood** [comment]
> CI is failing https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/232833493

This looks to be a flake. I can't reproduce this locally.

**Comment by brycekahle** [comment]
> This looks to be a flake

Does it consistently fail in CI?

**Comment by guyarb** [comment]
> > This looks to be a flake
> 
> Does it consistently fail in CI?

no, it is flaky


> > CI is failing https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/232833493
> 
> This looks to be a flake. I can't reproduce this locally.

I know, even before we couldn't reproduce it locally (happened to me once)
it reproduced "very easily" in the CI

**Comment by guyarb** [comment]
@hmahmood how do you want to proceed?

**Comment by hmahmood** [comment]
> @hmahmood how do you want to proceed?

At this point it looks like we're failing on the `rhel-86` `runtime` variant of the tests for kafka. I plan on looking into this more this week.

**Review by guyarb — APPROVED** [review]

**Pull request merged**